### ISSUE TITLE
Preformatted text with language -> HTML may be broken

### ISSUE THREAD

**Issue by chipsenkbeil** [vimwiki/vimwiki#1116]
From both `v2.5` and `dev` branches, the output of preformatted text with a language doesn't appear to yield proper HTML output:

```vimwiki
{{{python
code with language
on multiple lines
}}}
```

becomes

```html
<pre python>
code with language
on multiple lines
</pre>
```

It's unclear to me if there was a JavaScript library that was intended to be used with this, but major players like [highlight.js](https://highlightjs.org/) expect this format instead:

```html
<pre>
    <code class="python">
        code with language
        on multiple lines
    </code>
</pre>
```

The output also doesn't include `<code>` tags, which I'm unsure if that was intentional or not.

@brennen and I were discussing this on IRC earlier. @Nudin, @tinmarino, are either of you aware of what the output is supposed to be? Also, language aside, any use of preformatted text (with nothing, a language, or something like `class="brush: python"`) doesn't include `<code>`.

**Comment by brennen** [comment]
> It's unclear to me if there was a JavaScript library that was intended to be used with this

Docs mention "Google's syntax highlighter", but offhand I don't know exactly what that means.

Looks like [syntaxhighlighter](https://github.com/syntaxhighlighter/syntaxhighlighter) [supports](https://github.com/syntaxhighlighter/syntaxhighlighter/wiki/Usage) the `class="brush: foo"` variant also mentioned in the docs, so that's probably a clue.

Ideally I think the `{{{lang ... }}}` syntax would probably get converted to something appropriate by whatever renderer.

**Comment by chipsenkbeil** [comment]
Maybe Google Prettify, which has been discontinued: https://meta.stackexchange.com/questions/353983/goodbye-prettify-hello-highlight-js-swapping-out-our-syntax-highlighter

Depending on the highlighter, you'd need an option to configure the output on pre or code if it's Javascript rendering.

**Comment by tinmarino** [comment]
The output should be as you said. I remember to have forgotten the code tag and broke this behaviour in the commit dc63a5d  referred by #1007

**Comment by tinmarino** [comment]
sorry ...

**Comment by chipsenkbeil** [comment]
@tinmarino, that's referencing blockquote. This is about preformatted text (aka code blocks). Does your commit also affect preformatted text?

**Comment by tinmarino** [comment]
I think the tiple-backslash in mardown correspond to the triple {{{ in
vimwiki, no ?
I just gave the example in markdown

**Comment by chipsenkbeil** [comment]
Yeah, three backticks is code in markdown, but that also isn't a block quote. These are two separate issues, I think.

For code blocks, it's producing the above broken tags where it's missing `<code>` and isn't correctly translating the language on the tag, or maybe is but it isn't clear what front-end Javascript library is compatible. 

For block quotes in the other issue you referenced, it got changed from the `<blockquote>` tag to `<pre><code>`, which also isn't right.

**Comment by tinmarino** [comment]
Chip: I see, I ignored all that and hence the bug. I never use blockquotes.

I guess the code block issue is the more problematic (to get the language
in the tag).
